File: src/jvmception/objects/cp/CpMethodHandle.java
package jvmception.objects.cp;

import java.io.DataInputStream;
import java.io.IOException;

import jvmception.jvmtypes.IUnitSerializable;
import jvmception.jvmtypes.JVMReference;

public class CpMethodHandle extends CpInfo {
	private int referenceKind;
	private int referenceIndex;
	
	public CpMethodHandle(DataInputStream dis) throws IOException{
		referenceKind = dis.readUnsignedByte();
		referenceIndex = dis.readUnsignedShort();
	}
	
	public int getRefKind() {
		return referenceKind;
	}
	
	public int getRefIndex() {
		return referenceIndex;
	}

	@Override
	public IUnitSerializable toIUnitSerializable() {
		return JVMReference.NULL_REFERENCE;
	}
}

File: src/jvmception/objects/cp/CpMethodHandle.cs


using System.IO;

public class CpMethodHandle : CpInfo
{
    private int referenceKind;
    private int referenceIndex;

    public CpMethodHandle(BinaryReaderBigEndian dis)
    {
        referenceKind = dis.ReadByte();
        referenceIndex = dis.ReadUInt16();
    }

    public int getRefKind()
    {
        return referenceKind;
    }

    public int getRefIndex()
    {
        return referenceIndex;
    }

    public override IUnitSerializable toIUnitSerializable()
    {
        return JVMReference.NULL_REFERENCE;
    }
}
